frontend/src/pages/Registration.tsx

```diff
@@ -127,13 +129,25 @@ export const Registration: React.FC = (): JSX.Element => {
             <FormControl isRequired marginTop={6}>
               <FormLabel>{formatUserDetailsKey.password}</FormLabel>
               <InputGroup>
-                <Input name="password" type={showPassword ? "text" : "password"} placeholder="*******" size="md" />
+                <Input
+                  name="password"
+                  type={showPassword ? "text" : "password"}
+                  placeholder="*******"
+                  size="md"
+                  onChange={(event) => setPassword(event.target.value)}
+                />
                 <InputRightElement width="4.5rem">
                   <Button h="1.75rem" size="sm" onClick={handlePasswordShow} variant="ghost">
                     {showPassword ? <AiFillEyeInvisible /> : <AiFillEye />}
                   </Button>
                 </InputRightElement>
               </InputGroup>
+              <PasswordChecklist
```

Comment:
```suggestion
              </InputGroup>
              {getUserDetailsErrorMessage(errorMap, "password")}
              <PasswordChecklist
```
TODO: spacing between error message and password constraints list